### ISSUE TITLE
E3601 in `States/ProcessWorkItems/ResultWriter` - Expecting `string` instead of object

### ISSUE THREAD

**Issue by Doryanni** [aws-cloudformation/cfn-lint#4496]
### CloudFormation Lint Version

1.50.1

### What operating system are you using?

Arch Linux

### Describe the bug

When running `cfn-lint` over my template, I get an E3601 error in `States/ProcessWorkItems/ResultWriter`.
The linter is expecting this entity to be of type `string` whereas it is an object.

### Expected behavior

I would expect this entity to validate as an object, not a string, as seen [here](https://docs.aws.amazon.com/step-functions/latest/dg/input-output-resultwriter.html).

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"

Resources:
  StateMachineRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Version: "2012-10-17"
        Statement:
          - Effect: Allow
            Principal:
              Service: states.amazonaws.com
            Action: sts:AssumeRole
      Policies:
        - PolicyName: MinimalStepFunctionsPolicy
          PolicyDocument:
            Version: "2012-10-17"
            Statement:
              - Effect: Allow
                Action:
                  - s3:GetObject
                  - s3:PutObject
                  - s3:ListBucket
                  - states:StartExecution
                  - states:DescribeExecution
                  - states:StopExecution
                Resource: "*"

  ReproStateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      RoleArn: !GetAtt StateMachineRole.Arn
      Definition:
        StartAt: ProcessWorkItems
        States:
          ProcessWorkItems:
            Type: Map
            ItemReader:
              Resource: arn:aws:states:::s3:getObject
              ReaderConfig:
                InputType: JSONL
              Parameters:
                Bucket: example-input-bucket
                Key: input.jsonl
            ItemProcessor:
              ProcessorConfig:
                Mode: DISTRIBUTED
                ExecutionType: STANDARD
              StartAt: PassItem
              States:
                PassItem:
                  Type: Pass
                  End: true
            ResultWriter:
              Resource: arn:aws:states:::s3:putObject
              Parameters:
                Bucket: example-output-bucket
                Prefix: result-prefix
            End: true
```

**Comment by szbartnik** [comment]
https://github.com/aws-cloudformation/cfn-lint/blob/ce341bbf15c5c291293f44dbc05e309d5d662d7e/src/cfnlint/data/schemas/other/step_functions/statemachine.json#L616-L618

May be as easy as changing schema here to object. If it speeds up things, we are happy to create a PR
